server/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -771,6 +790,101 @@ private Set<RestRequest.Method> getValidHandlerMethodSet(String rawPath) {
         return validMethods;
     }
 
+    private static void recordRequestMetric(RestStatus statusCode, String handlerName, String requestMethod, LongCounter requestsCounter) {
+        try {
+            Map<String, Object> attributes = Map.of(
+                STATUS_CODE_KEY,
+                statusCode.getStatus(),
+                HANDLER_NAME_KEY,
+                handlerName,
+                REQUEST_METHOD_KEY,
+                requestMethod
+            );
+            requestsCounter.incrementBy(1, attributes);
+        } catch (Exception ex) {
+            logger.error("Cannot track request status code", ex);
+        }
+    }
+
+    private static void recordRequestMetric(RestStatus statusCode, LongCounter requestsCounter) {
+        try {
+            Map<String, Object> attributes = Map.of("es_rest_status_code", statusCode.getStatus());
+            requestsCounter.incrementBy(1, attributes);
+        } catch (Exception ex) {
+            logger.error("Cannot track request status code", ex);
+        }
+    }
+
+    private static final class MeteringRestChannelDecorator implements RestChannel {
```

Comment:
this shares same code with `ResourceHandlingHttpChannel` 
is it worth an abstract class?